src/components/faktum/FaktumDato.test.tsx

```diff
@@ -63,14 +61,244 @@ describe("FaktumDato", () => {
       );
 
       const datepicker = screen.getByLabelText(faktumMockData.beskrivendeId) as HTMLInputElement;
+      await user.type(datepicker, "04.08.2022");
 
-      // To trigger correct behavior as explained here: https://github.com/testing-library/user-event/issues/399#issuecomment-815664027
+      await waitFor(() => {
+        expect(onchange).toBeCalledTimes(1);
+        expect(onchange).toBeCalledWith(faktumMockData, "2022-08-04");
+      });
+    });
+  });
+
+  describe("When user removes an seleted date", () => {
+    test("Should post null to server", async () => {
+      faktumMockData.svar = "2022-08-04";
+      const user = userEvent.setup();
+      const onchange = jest.fn();
+
+      render(
+        <MockContext>
+          <FaktumDato faktum={faktumMockData} onChange={onchange} />
+        </MockContext>
+      );
+
+      const datepicker = screen.getByLabelText(faktumMockData.beskrivendeId) as HTMLInputElement;
       await user.clear(datepicker);
-      await user.type(datepicker, svar);
 
       await waitFor(() => {
         expect(onchange).toBeCalledTimes(1);
-        expect(onchange).toBeCalledWith(faktumMockData, svar);
+        expect(onchange).toBeCalledWith(faktumMockData, null);
+      });
+    });
+  });
+
+  describe("When user selects a date three weeks from now", () => {
```

Comment:
Få med at dette gjelder for det spesifikke tilfellet faktum.dagpenger-soknadsdato